src/local/room_position/game_methods.rs

```diff
@@ -0,0 +1,97 @@
+//! Game method implementations on `LocalRoomPosition`
+use crate::{
+    constants::{Color, FindConstant, LookConstant, ReturnCode, StructureType},
+    game,
+    macros::*,
+    objects::{FindOptions, Flag, HasPosition, LookResult, Path, RoomPosition},
+    pathfinder::CostMatrix,
+};
+
+use super::LocalRoomPosition;
+
+impl LocalRoomPosition {
+    pub fn create_construction_site(self, ty: StructureType) -> ReturnCode {
+        js_unwrap!(
+            pos_from_packed(@{self.packed_repr()})
+                .createConstructionSite(__structure_type_num_to_str(@{ty as u32}))
+        )
+    }
+
+    pub fn create_named_construction_site(self, ty: StructureType, name: &str) -> ReturnCode {
```

Comment:
Main reason is that `LocalRoomPosition` is the size of `u32`, and thus a reference to it will be the exact same size, and just require more indirection to use.

Since `LocalRoomPosition` is `Copy`, it should still work for `&LocalRoomPosition` - only difference is that no reference is needed.

I first thought about this when I encountered the [trivial_copy_pass_by_ref clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#trivially_copy_pass_by_ref).

Reply:
Oh that makes sense. I wasn't sure whether `Self` would also be taken by copy. Would it be worth changing the others also?

Reply:
Definitely! I, uh, thought I had. Must have missed that!